CONTRIBUTING.md

```diff
@@ -5,3 +5,43 @@ This will ensure we avoid duplicating work, or that your code can't be merged du
 base. If you would like support for a module that is not listed, [contact support][1] to share a request.
 
 [1]: https://docs.datadoghq.com/help
+
+## Keep changes small and incremental
+
+Changes should be incremental and understandable. As much as possible, large-scale efforts should be broken up into many PRs over time for better reviewability. If a feature would require more changes to be "complete" it's fine to land partial changes if they are not wired up to anything yet, so long as tests are included which at least prove those parts work in isolation.
+
+There are great benefits to taking a measured and iterative approach to improvement. When working on code in fewer places there is far less risk of running into merge conflicts or incompatibilities with other systems. Keeping contributions small makes them easy to review which makes that much quicker to land. Additionally, keeping things small and iterative makes it easier for other teams to review and understand what the code does.
+
+## Be descriptive!
+
+Sometimes code can be self-documenting, but often it can't. That is especially true to someone reviewing code they haven't worked on. If you're conscious of writing code in a self-describing way and leave comments anywhere that self-description fails you can go a long way to making even complex code coherent to one not already familiar with that code.
```

Comment:
I feel this is a bit of a run-on sentence. I'd suggest:
```suggestion
Sometimes code can be self-documenting, but often it can't. That is especially true to someone reviewing code they haven't worked on. Be conscious of writing code in a self-describing way and leave comments anywhere that self-description fails. This goes a long way towards making even complex code coherent to one not already familiar with it.
```
This also turns it into an actionable instruction ("Do this. Here's why.") instead of a hypothetical ("if you did this".)